configure.py

```diff
@@ -42,10 +42,78 @@
 from multiprocessing import Pool
 from subprocess import run
 
+_CPU_FAMILY_MAP = {
+    "arm": "arm",
+    "aarch64": "arm",
+    "x86_64": "x86",
+}
+
+_ANDROID_VERSION = 28
+
+_ANDROID_COMPILER_MAP = {
+    "arm": f"armv7a-linux-androideabi{_ANDROID_VERSION}",
+    "aarch64": f"aarch64-linux-android{_ANDROID_VERSION}",
+    "x86_64": f"x86_64-linux-android{_ANDROID_VERSION}",
+}
+
+_ANDROID_ABI_MAP = {
+    "arm": "armeabi-v7a",
+    "aarch64": "arm64-v8a",
+    "x86_64": "x86_64",
+}
```

Comment:
I think I'd prefer something like:
```python
@dataclass
class _Android:
    cpu_family: str
    compiler: str
    abi: str


_ANDROID_INFO = dict(
    arm=_Android(
        cpu_family="arm",
        compiler=f"armv7a-linux-androideabi{_ANDROID_VERSION}",
        abi="armeabi-v7a",
    ),
    aarch64=_Android(
        cpu_family="arm",
        compiler=f"aarch64-linux-android{_ANDROID_VERSION}",
        abi="arm64-v8a",
    ),
    x86_64=_Android(
        cpu_family="x86",
        compiler=f"x86_64-linux-android{_ANDROID_VERSION}",
        abi="x86_64",
    ),
)
````

Reply:
The CPU family is also relevant for iOS.

Reply:
I've adjusted the code to use the dict() function. But I'm not sure if it's worth having a dataclass for the compiler and abi field. What do you think ?

Reply:
That's fine for now, but I think it will make sense with the introduction of iOS. Like, maybe something like that:

```python
@dataclass
class _CompilerABI:
    compiler: str
    abi: str

@dataclass
class _CrossPlatform:
    cpu_family: str
    android: _CompilerABI
    ios: _CompilerABI

_CROSS_PLATFORM_INFO = dict(
    arm=_CrossPlatform(
        cpu_family="arm",
        android=_CompilerABI(f"armv7a-linux-androideabi{_ANDROID_VERSION}", "armeabi-v7a"),
        ios=_CompilerABI(...),
    ),
    aarch64=_CrossPlatform(
        cpu_family="arm",
        android=_CompilerABI(f"aarch64-linux-android{_ANDROID_VERSION}", "arm64-v8a"),
        ios=_CompilerABI(...),
    ),
    x86_64=_CrossPlatform(
        cpu_family="x86",
        android=_CompilerABI(f"x86_64-linux-android{_ANDROID_VERSION}", "x86_64"),
        ios=_CompilerABI(...),
    ),
)
```